azure-samples/src/main/java/com/microsoft/azure/management/resources/samples/DeployUsingARMTemplateWithProgress.java

```diff
@@ -145,7 +145,7 @@ private static void validateAndAddFieldValue(String type, String fieldValue, Str
         final ObjectMapper mapper = new ObjectMapper();
         final ObjectNode parameter = mapper.createObjectNode();
         parameter.put("type", type);
-        if (type == "int") {
```

Comment:
Fixing incorrect string comparison

Reply:
[P3]: This and the next 3 changes, the behavior of `type == "int"` and `"int".equals(type)` are same. Fine with using either approach.

Reply:
Seems `==` tests for reference equality so we should use `"int".equals(type)`

Reply:
I marking it as P2 as it is a fix needed in sample. Thanks for finding this.